apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/DiskLruHttpCache.kt

```diff
@@ -144,20 +143,16 @@ class DiskLruHttpCache(private val fileSystem: FileSystem, private val directory
     }
 
     override fun close() {
-      closeAndCommitCache()
-      originalSource.close()
-    }
-
-    private fun closeAndCommitCache() {
       if (!hasClosedAndCommitted) {
         try {
           sink.close()
-          cacheEditor.commit()
+          cacheEditor.commitUnlessAborted()
         } catch (e: Exception) {
           // Silently ignore cache write errors
         } finally {
           hasClosedAndCommitted = true
         }
+        originalSource.close()
```

Comment:
Should that be in a try/catch as well? I think it can throw

Reply:
If we successfully read the source, but then when trying to close it it threw, I think we should still commit? If yes, then no need to catch it, it will throw after the commit (or abort if there was an error).

Reply:
Right thanks for the deep dive 👍 . Sometimes I miss checked exceptions...